examples/scala/src/main/scala/example/IcebergCompatV2.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package example
+
+import java.io.{File, IOException}
+import java.net.ServerSocket
+
+import org.apache.commons.io.FileUtils
+
+import org.apache.spark.sql.SparkSession
+/**
+ * This example relies on an external Hive metastore (HMS) instance to run.
+ *
+ * A standalone HMS can be created using the following docker command.
+ *  ************************************************************
+ *  docker run -d -p 9083:9083 --env SERVICE_NAME=metastore \
+ *  --name metastore-standalone apache/hive:4.0.0-beta-1
+ *  ************************************************************
+ *  The URL of this standalone HMS is thrift://localhost:9083
+ *
+ *  By default this hms will use `/opt/hive/data/warehouse` as warehouse path.
+ *  Please make sure this path exists or change it prior to running the example.
+ */
+object IcebergCompatV2 {
+
+  def main(args: Array[String]): Unit = {
+    // Update this according to the metastore config
+    val port = 9083
```

Comment:
you can use [this](https://stackoverflow.com/questions/31277943/how-would-i-choose-in-java-that-which-port-is-free-to-use-for-serversocket-or-so) method to find the free port.

Reply:
Oh, we are not looking for the free port. This port must be the one used by Hive MetaStore. The user should start a Hive MetaStore using the docker command in the comments, then update the port here.